tests/integration/sql_session_manager.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import asdict, dataclass
+from typing import Any, Type
+
+import sqlalchemy as sa
+from sqlalchemy.pool import Pool, QueuePool
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class ConnectionDetails:
+    connection_string: str
+    # BDIRKS extract this from connection string
+    dialect: str
+
+
+@dataclass(frozen=True)
+class PoolConfig:
+    poolclass: Type[Pool]
+    pool_size: int
+    max_overflow: int
+    pool_recycle: int
+    pool_timeout: int
+    pool_pre_ping: bool
+
+
+class SessionSQLEngineManager:
```

Comment:
This is the new session manager for tests. This and the fixture that uses it are the non-plumbing changes in this PR.